jq255.md

```diff
@@ -0,0 +1,1405 @@
+    verification is also faster than with the common 64-byte signatures.
+
+  - A key exchange protocol (Diffie-Hellman).
+
+  - Map-to-curve and hash-to-curve procedures, that allow one-way
+    mapping of arbitrary data into group elements with purely
+    constant-time implementations.
+
+The underlying curves are specific double-odd curves (curves whose order
+is twice an odd integer) interpreted with a coordinate system such that
+the curve equation is a Jacobi quartic, as described in the
+[double-odd Jacobi quartic whitepaper]. More information on double-odd
+curves can be found on the [double-odd elliptic curves Web site].
+
+## Notations
+
+  - `x^y` denotes exponentiation of `x` by the integer `y`. This is
+    applicable to any type of `x` for which a multiplication operation
+    is defined (e.g. integers or finite field elements).
+
+  - Each group is defined over the *finite field* `GF(q)` consisting of
+    integers modulo `q = 2^255 - mq` for some small integer `mq` such
+    that `q` is prime. For both jq255e and jq255s, `mq < 2^15` so that
+    implementations on small microcontrollers remain efficient.
+
+  - The *sign* of a field element `x`, denoted `sgn(x)`, is defined as
+    the least significant bit of the binary representation of `x` as
+    an integer in the `0` to `q-1` range. Thus, for any `x`:
+
+      - `sgn(x)` is equal to `0` or `1`.
+      - If `x = 0`, then `sgn(x) = sgn(-x) = 0`.
+      - If `x != 0`, then `sgn(x) = 1 - sgn(-x)`.
+
+    A value `x` is said to be *negative* if `sgn(x) = 1`, or
+    *non-negative* if `sgn(x) = 0`.
+
+  - The underlying elliptic curve contains exactly `n = 2*r` points,
+    with `r` being a prime integer. The group itself (jq255e or jq255s)
+    has order `r`. For both groups, `r` is close to `2^254`.
+
+  - The elliptic curve is the set of points `(e, u)`, with both
+    coordinates being elements of `GF(q)` that fulfill the curve
+    equation `e^2 = (a^2 - 4*b)*u^4 - 2*a*u^2 + 1`, for two constants
+    `a` and `b` that are part of the curve definition. The curve neutral
+    point (the "point-at-infinity" which is not actually "at infinity"
+    in this coordinate system) is `I = (1,0)`. The curve also contains a
+    single point of order 2 denoted `N = (-1,0)`.
+
+  - For ease of notation, we define `a' = -2*a` and `b' = a^2 - 4*b`.
+    With these constants, the curve equation becomes:
+    `e^2 = b'*u^4 + a'*u^2 + 1`
+
+  - A *group element* is defined as a pair `{ P, P+N }` for any curve
+    point `P`. The points `P` and `P+N` are the two possible
+    *representants* of the group element. The group law is denoted
+    additively: for any two group elements, their sum is computed by
+    simply adding together their representant points over the curve;
+    since `N` has order 2, it does not matter which representant is used
+    for each group element. The encoding process ensures that the same
+    output is obtained for both representants of the same group element.
```

Comment:
I would tweak this wording to indicate that while the author's/encoding's choice of which of the two points to represent the group element doesn't matter, there _is_ one correct canonical representation of the group element, and doing funny stuff with the other point is a bad idea and abstraction violation that may lead to tears

Reply:
Wording has been tweaked.